ui/src/app/views/new-dataset-modal.tsx

```diff
@@ -161,9 +160,11 @@ class NewDataSetModal extends React.Component<Props, State> {
             notebookName: this.state.notebookName,
             newNotebook: this.state.newNotebook
           });
-        navigate(['workspaces',
-          workspaceNamespace,
-          workspaceId, 'notebooks', this.state.notebookName + '.ipynb']);
+        // Open notebook in a new tab and return back to the Data tab
+        const notebookUrl = '\\workspaces\\' + workspaceNamespace + '\\' + workspaceId +
+          '\\notebooks\\' + this.state.notebookName + '.ipynb';
+        window.history.back();
+        window.open(notebookUrl);
```

Comment:
Nit: Ordering-wise I would probably open the notebook URL before going back in the history.